code/modules/forensics/_forensics.dm

```diff
@@ -0,0 +1,235 @@
+/**
+ * Forensics datum
+ *
+ * Placed onto atoms, and contains:
+ * * List of fingerprints on the atom
+ * * List of hidden prints (used for admins)
+ * * List of blood on the atom
+ * * List of clothing fibers on the atom
+ */
+/datum/forensics
+	/// Weakref to the parent owning this datum
+	var/datum/weakref/parent
+	/**
+	 * List of fingerprints on this atom
+	 *
+	 * Formatting:
+	 * * print = print
+	 */
+	var/list/fingerprints
+	/**
+	 * List of fingerprints on this atom
+	 *
+	 * Formatting:
+	 * * ckey = realname/gloves/ckey
+	 */
+	var/list/hiddenprints
+	/**
+	 * List of fingerprints on this atom
+	 *
+	 * Formatting:
+	 * * dna = bloodtype
+	 */
+	var/list/blood_DNA
+	/**
+	 * List of clothing fibers on this atom
+	 *
+	 * Formatting:
+	 * * fiber = fiber
+	 */
+	var/list/fibers
+
+/datum/forensics/New(atom/parent, fingerprints, hiddenprints, blood_DNA, fibers)
+	if(!isatom(parent))
+		stack_trace("We tried adding a forensics datum to something that isnt an atom. What the hell are you doing?")
+		qdel(src)
+		return
+
+	RegisterSignal(parent, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_act)
+
+	src.parent = WEAKREF(parent)
+	src.fingerprints = fingerprints
+	src.hiddenprints = hiddenprints
+	src.blood_DNA = blood_DNA
+	src.fibers = fibers
+	check_blood()
+
+/// Merges the given lists into the preexisting values
+/datum/forensics/proc/inherit_new(list/fingerprints, list/hiddenprints, list/blood_DNA, list/fibers) //Use of | and |= being different here is INTENTIONAL.
+	if (fingerprints)
+		src.fingerprints = LAZY_LISTS_OR(src.fingerprints, fingerprints)
+	if (hiddenprints)
+		src.hiddenprints = LAZY_LISTS_OR(src.hiddenprints, hiddenprints)
+	if (blood_DNA)
+		src.blood_DNA = LAZY_LISTS_OR(src.blood_DNA, blood_DNA)
+	if (fibers)
+		src.fibers = LAZY_LISTS_OR(src.fibers, fibers)
+	check_blood()
+
+/datum/forensics/Destroy(force, ...)
+	UnregisterSignal(parent.resolve(), list(COMSIG_COMPONENT_CLEAN_ACT))
```

Comment:
as parent's a weakref, you may need to check that this resolve is successful when it's done.
in other cases, this'll probably never come into play, but in `Destroy()` all bets are off. 
```
var/atom/real_parent = parent.resolve()
if(real_parent)
   ...
```

Reply:
gotcha